### ISSUE TITLE
grok URI pattern is broken in some cases

### ISSUE THREAD

**Issue by hrchu** [logstash-plugins/logstash-patterns-core#57]
Fail case: http://www.yahoo.com/SDK_%E9%96%8B%E7%99%BC%E6%8C%87%E5%8D%97.zip0&Expires=1433136894

The result is http://www.yahoo.com/SDK_%E9%96%8B%E7%99%BC%E6%8C%87%E5%8D%97.zip0, lose Expires=1433136894

Tested by https://grokdebug.herokuapp.com/

**Comment by markwalkom** [comment]
Neither of those Yahoo links work.

**Comment by hrchu** [comment]
The url is just for parsing test. Does this effect the result?

**Comment by markwalkom** [comment]
Yes. We have no information to even discuss what your problem is.

**Comment by purbon** [comment]
Hi @hrchu your url looks strange to me, why do you have all this encoded characters? do you have a valid working url as example? like this we would be able to reproduce this bug and address the issue where need if possible.

Thanks a lot for your time,
